### PR TITLE
Added helm upgrade strategy 'delete' to nvidia addon

### PR DESCRIPTION
**Pull request by Gilbert88**
**What type of PR is this?**
Bug

**What this PR does/ why we need it**:
This PR fixes the Nvidia GPU addon ugprade issue.

**Which issue(s) this PR fixes**:
[D2IQ-64166](https://jira.d2iq.com/browse/D2IQ-64166)

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
NONE

**Checklist**

* [x] The commit message explains the changes and why are needed.
* [x] The code builds and passes lint/style checks locally.
* [ ] The relevant subset of integration tests pass locally.
* [ ] The core changes are covered by tests.
* [x] The documentation is updated where needed.

### CODE REVIEW

File: addons/nvidia/0.3.x/nvidia-2.yaml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+apiVersion: kubeaddons.mesosphere.io/v1beta1
+kind: ClusterAddon
+metadata:
+  name: nvidia
+  namespace: kubeaddons
+  labels:
+    kubeaddons.mesosphere.io/name: nvidia
+    kubeaddons.mesosphere.io/provides: nvidia
+  annotations:
+    catalog.kubeaddons.mesosphere.io/addon-revision: "0.3.0-2"
```
Comment (shaneutt) [inline addons/nvidia/0.3.x/nvidia-2.yaml]: The revision number should reference the app number. What is this semver referencing?

Comment (Gilbert88) [reply addons/nvidia/0.3.x/nvidia-2.yaml]: Updated to 0.2.0-2 and rebased to master

File: addons/nvidia/0.3.x/nvidia-2.yaml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+kind: ClusterAddon
+metadata:
+  name: nvidia
+  namespace: kubeaddons
+  labels:
+    kubeaddons.mesosphere.io/name: nvidia
+    kubeaddons.mesosphere.io/provides: nvidia
+  annotations:
+    catalog.kubeaddons.mesosphere.io/addon-revision: "0.2.0-2"
+    appversion.kubeaddons.mesosphere.io/nvidia: "0.2.0"
```
Comment (GoelDeepak) [inline addons/nvidia/0.3.x/nvidia-2.yaml]: I am a bit confused why version `0.2.0` is in `0.3.x` directory?

File: addons/nvidia/0.3.x/nvidia-2.yaml (hunk 1 above)
Comment (joejulian) [reply addons/nvidia/0.3.x/nvidia-2.yaml]: What does this mean, @shaneutt ? The last one was 0.3.0-1 should this not be 0.3.0-2?

btw, the links are broken: https://github.com/mesosphere/kubernetes-base-addons#overview

**Comment by joejulian** [comment]
Please re-approve. I had to recreate the branch because someone didn't sign their commits.

**Pull request merged**